Submission of substrate_client_java

strategyobjectadmin opened the pull request:
Correction of typos and project name

Ready for submission

### Project Abstract

> Please provide a brief description of your project here summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [X] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [X] The software delivered for this grant will be released under an open-source license specified in the application.
- [X] The initial PR contains only one commit (squash and force-push if needed).
- [X] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Grants-Program?pullRequest=726) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Grants-Program?pullRequest=726) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Grants-Program?pullRequest=726) it.</sub>

semuelle commented:
Thank you for your application, @strategyobjectadmin! We will look into it as soon as possible.

Noc2 reviewed (changes requested):
Thanks for the application. Are you aware of [PolkaJ](https://github.com/emeraldpay/polkaj)? It is not super well maintained, but it probably makes sense to leverage this for your project.  There is also https://github.com/polkadot-java/api (which is outdated at the moment).

strategyobjectadmin commented:
Thanks for the tip @Noc2 we've added a clarification on this into the application form. Actually, we've researched the available solutions before writing one of our own. Up to now we've already implemented the client supporting features that are available in the libraries that you mentioned and are using it in our corporate landscape (available [here](https://github.com/strategyobject/substrate-client-java)). Our application is not about creating a new client from scratch, but is more about continuing further development. We believe that the architectural approach that we picked will lead to a fully functional, easy to use and maintain solution.

Noc2 commented:
Thanks for the reply here. I have one additional comment: The license seems to be missing for all of your deliveries and in your repository. Is this on purpose? This is usually mandatory for all milestones, see our template:  https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules

strategyobjectadmin commented:
No, that wasn't on purpose. The license is added now to both the repository and the application form.

gautamdhameja commented:
Would be good to make it compatible with Android too.

strategyobjectadmin commented:
@gautamdhameja indeed. We have this in mind since we are currently designing a mobile app that we'd like to let communicate with the substrate network directly. But firstly we'd like to concentrate on the main goal and leave it for later.

semuelle commented:
@strategyobjectadmin @gautamdhameja We have an ongoing grant to make PolkaJ Android compatible: https://github.com/emeraldpay/polkaj/pulls?q=is%3Apr+Android & https://github.com/w3f/Grants-Program/pull/301

alxs commented on applications/substrate_client_java.md:
```diff
@@ -0,0 +1,409 @@
+- **Project Name:** Substrate Client for Java
+- **Team Name:** Doken  
+- **Payment Address:** Information provided in the "W3F Technical Grant Application Form"  
+- **Level:** 3
```
I suggest removing this - no need to specify it if the costs are kept private.

```suggestion
```

alxs commented on applications/substrate_client_java.md:
```diff
@@ -0,0 +1,408 @@
+### Legal Structure
+
+- **Registered Address:** ul. "Tsaribrodska" 70, 1309 Sveta Troitsa, Sofia, Bulgaria
+- **Registered Legal Entity:** Strategy Object Ltd.
```
```suggestion
- **Registered Legal Entity:** Strategy Object EOOD
```

alxs commented:
Hey @strategyobjectadmin, what are your plans for milestone 3? It looks like there's been [no activity](https://github.com/strategyobject/substrate-client-java/graphs/contributors) on the project repository since the last delivery.

strategyobjectadmin commented:
Hey @alxs Sorry, for the delay. The team was very busy with the other tasks. Will continue to work on the client asap.

strategyobjectadmin commented:
Hi @keeganquigley. Sorry for the delay. We will make the amendment during next week.

keeganquigley commented:
Hi @strategyobjectadmin just a friendly reminder as I haven't seen an amendment filed yet. Thanks!

keeganquigley commented:
Hi @strategyobjectadmin are you still working on this grant?

keeganquigley commented:
Hi @strategyobjectadmin since we haven't heard from you after repeated attempts, we will go ahead and close the grant for now. Let us know if you wish to continue working on it.

strategyobjectadmin commented:
Hello,
The Java Substrate library is actively in use in production. However, we
have been internally discussing adjustments to our roadmap, as our current
production engagements are taking considerable time.
Please allow us some additional time to redefine our roadmap, and we will
get back to you with a clear action plan.
Thank you for your understanding.
Best Regards,
Andrea

On Fri, Nov 8, 2024 at 1:53 AM 'Keegan | W3F' via
***@***.*** ***@***.***> wrote:

> Hi @strategyobjectadmin <https://github.com/strategyobjectadmin> since we
> haven't heard from you after repeated attempts, we will go ahead and close
> the grant for now. Let us know if you wish to continue working on it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/726#issuecomment-2463302672>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWFTNV3XKVFBEF6PWZIZFRLZ7POMVAVCNFSM6AAAAABQRWBNHSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRTGMYDENRXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley commented:
Thanks @strategyobjectadmin just replied by email.

keeganquigley commented:
Grant has been canceled per email thread.
